src/toil_scripts/lib/__init__.py

```diff
@@ -31,6 +35,16 @@ def partitions(l, partition_size):
         yield l[i:i + partition_size]
 
 
+def get_work_directory(env_variable='TOIL_WORKDIR', backup_dir='/mnt/ephemeral'):
```

Comment:
This method should create a uniquely named temporary subdirectory in the first two cases.

s/backup/fallback otherwise it sounds like a directory for making backups.

I need to be convinced me of the value of being able to override `env_variable` and `backup_dir`/`fallback_dir`. We may want to tweak the implementation in the future and a complicated signature would make these tweaks harder.